### ISSUE TITLE
Provide Input Gradient

### ISSUE THREAD

**Issue by hsidky** [OpenANN/OpenANN#179]
Hi!

Currently, Net provides the gradient of the loss function with respect to the parameters (understandably so). I was wondering if it is possible to obtain the gradient of an optimized network output with respect to an input vector. So given `X_i` as an input vector and `Y_i = net(X_i)` as the network prediction output vector, can I obtain `dY_i/dX_j` ?

Thanks!

**Comment by AlexanderFabisch** [comment]
Hi,

unfortunately that is not implemented. You can calculate for example dE/dX easily because that is already available [here](https://github.com/OpenANN/OpenANN/blob/master/OpenANN/layers/Layer.h#L72) although it is not exposed by the Net.

If you need dY/dX I would recommend to use either Theano or Tensorflow because these libraries provide automatic differentiation.

**Comment by hsidky** [comment]
I see. Thanks for your response. I quite like the OpenANN C++ API, so I will work on this myself.

**Comment by AlexanderFabisch** [comment]
Would you send me a pull request if you have something useful?
